tests/svs/core/distances/cosine.cpp

```diff
@@ -82,20 +82,39 @@ void test_types(T lo, T hi, size_t num_tests) {
                             .epsilon(COSINE_EPSILON)
                             .margin(COSINE_MARGIN);
 
-        // Statically Sized Computation
+        auto as_span = [](auto v) { return std::span{v.data(), v.size()}; };
+        auto& arch_env = svs::arch::MicroArchEnvironment::get_instance();
+        auto supported_archs = arch_env.get_supported_microarchs();
+
+        // Check the distance computation for all supported architectures.
+        for (auto arch : supported_archs) {
+            arch_env.set_microarch(arch);
+            auto dist = svs::distance::DistanceCosineSimilarity{};
+            dist.fix_argument(as_span(a));
+            auto result = svs::distance::compute(dist, as_span(a), as_span(b));
+            CATCH_REQUIRE(result == expected);
```

Comment:
Easier to read along with next assertions.
```suggestion
            auto dist_type = svs::distance::DistanceCosineSimilarity{};
            dist_type.fix_argument(as_span(a));
            auto dist = svs::distance::compute(dist_type, as_span(a), as_span(b));
            CATCH_REQUIRE(dist == expected);
```